src/udev/net/link-config.c

```diff
@@ -937,6 +938,61 @@ static int link_apply_sr_iov_config(Link *link, sd_netlink **rtnl) {
         return 0;
 }
 
+static int link_apply_rps_cpu_mask(Link *link) {
+        _cleanup_free_ char *mask_str, *dev_path, *rps_cpus_path = NULL;
+        _cleanup_closedir_ DIR *d = NULL;
+        LinkConfig *config;
+        int r;
+
+        assert(link);
+        config = ASSERT_PTR(link->config);
+
+        mask_str = cpu_set_to_mask_string(&config->rps_cpu_mask);
+        if (!mask_str)
+                return -ENOMEM;
+
+        log_link_debug(link, "Applying RPS CPU mask: %s", mask_str);
+
+        dev_path = path_join("/sys/class/net/", link->ifname, "queues");
```

Comment:
please start from the sd_device object that you find in link->device, do not make up paths on your own.

Reply:
and use FOREACH_DEVICE_CHILD_WITH_SUFFIX or so, like the iov stuff does